core/bind/core_bind.cpp

```diff
@@ -2520,10 +2523,10 @@ bool _Directory::dir_exists(String p_dir) {
 	}
 }
 
-int _Directory::get_space_left() {
+uint64_t _Directory::get_space_left() {
 
 	ERR_FAIL_COND_V_MSG(!d, 0, "Directory must be opened before use.");
-	return d->get_space_left() / 1024 * 1024; //return value in megabytes, given binding is int
+	return d->get_space_left() / 1024 * 1024; // Truncate to closest MiB.
```

Comment:
I think this was a misunderstanding / the comment was bogus. This does convert bytes to megabytes, it truncates to the nearest MiB and then (supposedly) converts back to KiB.

That being said, the method seems bogus on Linux at least, but that's a separate issue.

Reply:
Heh interesting, the Linux 64-bit implementation seems to be broken.

On my system:
```
/dev/nvme0n1p6   59G   46G   11G  82% /
/dev/nvme0n1p7  196G  172G   23G  89% /home
```

If I print the `get_space_left()` for `/` and `/home`, I get:

32-bit:
```
14093373440
25898131456
```
Checks out taking into account reserved block count I guess? Assuming the value is in bytes and not MiB as the comment suggests.

64-bit:
```
1208475648
222502912
```
what?

Reply:
Also broken on macOS (it's the same shared Unix code):

```
> df -H
Filesystem          Size   Used  Avail Capacity   iused      ifree %iused  Mounted on
/dev/disk1s5s1      250G    15G    18G    47%    568975 2442362825    0%   /
...
/dev/disk2s2        107G    52G    55G    49%    492742 4294474553    0%   /Volumes/Windows 10
...
```

```
> get_space_left()
8983348772864 for "/"
8983348772864 for "/Volumes/Windows 10"
```

Reply:
I opened #47262 to keep track of it.